docs/SETUP_TLS_CERTIFICATES.md

```diff
@@ -0,0 +1,232 @@
+    d=frodo.buthey.net; p=$(sudo find . -wholename "*live/$d"); for f in $(sudo ls -1 $p |grep .pem); do sudo cp $p/$f $(basename $p)-$f; done; sudo chown $(whoami) $d-*.pem
+    
+    ```
+    Resulting files:
+    * `frodo.buthey.net-privkey.pem` - private key file for domain `frodo.`,
+    * `frodo.buthey.net-cert.pem` - certificate file for domain `frodo.`,
+    * `frodo.buthey.net-chain.pem` - a certificate chain representing Let's encrypt CA (a certificate root).
+    
+## Installing certificates in system (Ubuntu, CentOS)
+
+This guide shows how to install root CA certificates in system. Having root CA certificates installed in system reduces complexity of TLS setup.
+
+### Installing root certificate in Ubuntu
+
+1. Obtain a certificate root file (e.g. `snaptest-CA.crt` from the guides in this document).
+1. Install the certificate in the system:
+    ```
+    sudo cp snaptest-ca.crt /usr/local/share/ca-certificates/
+    sudo update-ca-certificates --verbose --fresh
+    ```
+    Sample output:
+    ```
+    Clearing symlinks in /etc/ssl/certs...
+    done.
+    Updating certificates in /etc/ssl/certs...
+    Doing .
+    WARNING: Skipping duplicate certificate ACCVRAIZ1.pem
+    WARNING: Skipping duplicate certificate ACCVRAIZ1.pem
+    173 added, 0 removed; done.
+    ```
+
+### Installing root certificate in Centos
+
+1. Obtain a certificate root file (e.g. `snaptest-ca.crt` from the guides in this document).
+1. Install the certificate tooling and certificate in the system:
+    ```
+    yum install ca-certificates
+    update-ca-trust enable
+    cp /opt/snap/certs/snaptest-ca.crt /etc/pki/ca-trust/source/anchors/
+    update-ca-trust extract
+    ```
+
+## Testing TLS-secured plugin communication in local setup
+
+We'll assume the following files are all available:
+* `snaptest-ca.crt` - certificate root - installed in system,
+* `snaptest-srv.key` - server's private key file,
+* `snaptest-srv.crt` - server's certificate file,
+* `snaptest-cli.key` - framework's private key file,
+* `snaptest-cli.crt` - framework's certificate file.
+
+### Starting a telemetry workflow:
+
+1. Start new shell session and execute `snapteld` with certificate and key generated for client:
+    ```
+    snapteld -t 0 -l 1 --tls-cert=/tmp/snaptest-cli.crt --tls-key=/tmp/snaptest-cli.key
+    ```
+
+1. Load a plugin
+
+    Start new shell session and load plugin built with [snap-plugin-lib-go](https://github.com/intelsdi-x/snap-plugin-lib-go), e.g. `snap-plugin-collector-rand` from `snap-plugin-lib-go` examples:
+    ```
+    snaptel plugin load --plugin-cert=/tmp/snaptest-srv.crt --plugin-key=/tmp/snaptest-srv.key snap-plugin-collector-rand
+    ```
+    
+1. Start a sample task and watch the output:
+    ```
+    snaptel task create -t sample-task.json
+    
+    snaptel task watch 167384ad-7643-43ef-930a-18455f5307a0
```

Comment:
Nevermind,  the suggestion to pivot to the Sample output below is way better. Ignore repeating yourself like I suggested above.